modin/experimental/core/execution/native/implementations/omnisci_on_native/io/io.py

```diff
@@ -466,13 +466,14 @@ def _read_csv_check_support(
             )
 
         parse_dates_unsupported = isinstance(parse_dates, dict) or (
-            isinstance(parse_dates, list) and isinstance(parse_dates[0], list)
+            isinstance(parse_dates, list)
+            and any(not isinstance(date, str) for date in parse_dates)
```

Comment:
I'm curious why only dates that are string column names are supported and not int

e.g. `parse_dates = [1, 2, 3]`

is this something specific to the arrow engine?

Reply:
It seems pyarrow doesn't support `parse_dates`, so we have to [manually convert](https://github.com/mvashishtha/modin/blob/ca8e7b5955527abbe0c7927febbe29d205d7daff/modin/experimental/engines/omnisci_on_ray/io.py#L147) to timestamps ourselves. We haven't implemented support for the full variety of parse date types on omnisci. I think supporting `parse_dates` in OmniSci is beyond the scope of this PR.